### DISCUSSION TITLE
Queries without session instance

### DISCUSSION THREAD

**Discussion by fleeser** [serverpod/serverpod#1609] [Q&A] [answered]
I implemented a Firebase Cloud Messaging Service and currently try to handle device token management. Therefore when an automatically sent push notification returns a specific error code that indicates that the token is no longer used, I'd like to remove it from the database. 

```dart
Map<String, dynamic> jsonMap = json.decode(res.body);
String? errorCode = jsonMap['error']?['details']?[0]?['errorCode'];

if (errorCode == 'UNREGISTERED') await FcmToken.delete(session, where: (t) => t.fcmToken.equals(deviceToken));
```

The problem here is that queries need a session to be passed which is not available since these are push notifications that aren't send by a user but the system itself.

Isn't it possible to make database queries without session?

**Comment by SandPod** [comment, marked as answer]
Hey @fleeser!

A session is always required to communicate with the database as it is responsible for establishing the connection. You can however create your own session by calling the `createSession(...)` method on the running Serverpod instance. Note that you always need to call close on any session that you have created.

The running Serverpod instance is accessible through `Serverpod.instance`.
